File: sharedLib/src/androidMain/kotlin/com/gaoyun/roar/domain/user/DeleteRemoteAccountExecutorImpl.kt
package com.gaoyun.roar.domain.user

import com.gaoyun.roar.util.Preferences
import com.gaoyun.roar.util.PreferencesKeys
import com.google.firebase.Firebase
import com.google.firebase.auth.auth
import com.google.firebase.storage.storage

class DeleteRemoteAccountExecutorImpl(private val preferences: Preferences) : DeleteRemoteAccountExecutor {
    override fun deleteAccount() {
        try {
            preferences.getString(PreferencesKeys.CURRENT_USER_ID)?.let {
                Firebase.storage.reference.child("sync_data/$it.json").delete()
            }
            Firebase.auth.currentUser?.delete()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}

File: iosApp/iosApp/app/DeleteRemoteAccountExecutorImpl.swift
import FirebaseAuth
import FirebaseStorage
import sharedLib

class DeleteRemoteAccountExecutorImpl: DeleteRemoteAccountExecutor
{
    let currentUserPreferenceKey = PreferencesKeys().CURRENT_USER_ID
    let provider: KoinProvider

    init(provider: KoinProvider) {
        self.provider = provider
    }


    func deleteAccount()
    {
        let storageRef = Storage.storage().reference()
        guard let userId = provider.preferences.getString(key: currentUserPreferenceKey)
        else { return }

        storageRef.child("sync_data/\(userId).json").delete{ _ in }
        Auth.auth().currentUser?.delete()
    }
}
